### ISSUE TITLE
Question about yoloworld v2 close set tuning on coco dataset

### ISSUE THREAD

**Issue by Kevinfringe** [ultralytics/ultralytics#17685]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi Team,

Recently, I'm facing an issue when tuning yoloworld v2 on COCO dataset. The mAP keeps at a very low level from the very first  epoch and no changing with the training going. Then I follow the example on ultralytics official website to do a quick testify on coco8 about the training setting and some other configurations that I might ignore, and the  same thing happened.

Below is the training log:
tensor([[[ 1.5072e-02,  7.6553e-03,  1.3599e-02,  ..., -2.3513e-02, -1.9081e-02,  1.3951e-03],
         [ 9.1072e-03, -1.3291e-02,  1.2951e-02,  ...,  1.2794e-02,  2.2071e-02, -2.3793e-02],
         [ 6.0329e-03,  9.7927e-03,  3.5331e-03,  ..., -2.2410e-02, -2.4775e-02,  1.0401e-02],
         ...,
         [ 5.3733e-05, -2.9611e-02, -3.6656e-03,  ..., -2.5125e-02, -4.5288e-02, -2.0687e-02],
         [-6.5240e-03,  5.7907e-03, -3.5351e-02,  ..., -1.6803e-02, -3.4719e-02, -7.0487e-03],
         [-2.2818e-02,  1.5974e-02, -3.1339e-02,  ..., -9.7308e-03, -4.8753e-02,  2.7138e-02]]])
Ultralytics 8.3.9 🚀 Python-3.9.12 torch-1.13.0+cu117 CUDA:0 (NVIDIA GeForce RTX 2080 Ti, 11012MiB)
WARNING ⚠️ Upgrade to torch>=2.0.0 for deterministic training.
engine/trainer: task=detect, mode=train, model=/yoloworld_ultralytics/ultralytics-main/yolov8_weights/yolov8m-worldv2.pt, data=/yoloworld_ultralytics/ultralytics-main/ultralytics_mn/cfg/datasets/coco8.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=/home/xa00397/NMQT_Tool_commit/ivot_ai_qat/runs/detect/train

                   from  n    params  module                                       arguments                     
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]                 
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]                
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]             
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]               
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]           
  5                  -1  1    664320  ultralytics.nn.modules.conv.Conv             [192, 384, 3, 2]              
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]           
  7                  -1  1   1991808  ultralytics.nn.modules.conv.Conv             [384, 576, 3, 2]              
  8                  -1  2   3985920  ultralytics.nn.modules.block.C2f             [576, 576, 2, True]           
  9                  -1  1    831168  ultralytics.nn.modules.block.SPPF            [576, 576, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  2   2498118  ultralytics.nn.modules.block.C2fAttn         [960, 384, 2, 192, 6]         
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  2    668451  ultralytics.nn.modules.block.C2fAttn         [576, 192, 2, 96, 3]          
 16                  15  1    332160  ultralytics.nn.modules.conv.Conv             [192, 192, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  2   2350662  ultralytics.nn.modules.block.C2fAttn         [576, 384, 2, 192, 6]         
 19                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  2   5267817  ultralytics.nn.modules.block.C2fAttn         [960, 576, 2, 288, 9]         
 22        [15, 18, 21]  1   4075222  ultralytics.nn.modules.head.WorldDetect      [80, 512, True, [192, 384, 576]]
YOLOv8m-worldv2 summary: 326 layers, 28,376,158 parameters, 28,376,142 gradients

Transferred 529/532 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLO11n...
Downloading https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt to 'yolo11n.pt'...
⚠️ Download failure, retrying 1/3 https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt...
##O#-  #                                                                                                                      
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443 
⚠️ Download failure, retrying 2/3 https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt...
##O#-  #                                                                                                                      
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443 
⚠️ Download failure, retrying 3/3 https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt...
##O#-  #                                                                                                                      
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443 
AMP: checks skipped ⚠️, offline and unable to download YOLO11n. Setting 'amp=True'. If you experience zero-mAP or NaN losses you can disable AMP with amp=False.
train: Scanning /yoloworld_ultralytics/ultralytics-main/data/coco8/labels/trai
val: Scanning /yoloworld_ultralytics/ultralytics-main/data/coco8/labels/val.ca
No module named 'seaborn'
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000119, momentum=0.9) with parameter groups 84 weight(decay=0.0), 95 weight(decay=0.0005), 101 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /runs/detect/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/100      3.23G      1.021      5.379      1.356         22        640: 100%|██████████| 1/1 [00:03<00:00,  3.29s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:39<00:00, 39
                   all          4         17     0.0714      0.467      0.267      0.098

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/100      2.49G      1.006      5.184      1.459         32        640: 100%|██████████| 1/1 [00:00<00:00,  1.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9
                   all          4         17     0.0833      0.467      0.263     0.0988

      ........

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     52/100      2.73G     0.5786      3.655      1.196         29        640: 100%|██████████| 1/1 [00:00<00:00,  4.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 27
                   all          4         17     0.0833     0.0167     0.0458     0.0413

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     53/100      2.74G     0.8777      4.366      1.326         11        640: 100%|██████████| 1/1 [00:00<00:00,  3.72it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 13
                   all          4         17     0.0833     0.0167     0.0458     0.0413

The validation result after training on coco8 is:

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 20.97it/s
                   all          4         17      0.111      0.467      0.272      0.113
                person          3         10      0.333        0.3       0.34      0.248
                   dog          1          1      0.111          1      0.995      0.199
                 horse          1          2          0          0          0          0
              elephant          1          2      0.111        0.5      0.099     0.0495
              umbrella          1          1      0.111          1      0.199      0.179
          potted plant          1          1          0          0          0          0

The code is as below:

    model = YOLOWorld('/yoloworld_ultralytics/ultralytics-main/yolov8_weights/yolov8m-worldv2.pt')
    print(model.model.txt_feats)
    metrics = model.train(data="/yoloworld_ultralytics/ultralytics-main/ultralytics/cfg/datasets/coco8.yaml",epochs=100, imgsz=640)`

Which is nearly identical with the code snippet on [official website](https://docs.ultralytics.com/models/yolo-world/#how-do-i-train-a-yolo-world-model-on-my-dataset)

Besides, the validation result on coco8 looks quite good 

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  6.29it/s]
                   all          4         17      0.634      0.867      0.867      0.693
                person          3         10      0.686        0.7      0.682      0.411
                   dog          1          1      0.902          1      0.995      0.895
                 horse          1          2      0.559          1      0.995      0.749
              elephant          1          2      0.577        0.5      0.537      0.215
              umbrella          1          1      0.532          1      0.995      0.995
          potted plant          1          1      0.546          1      0.995      0.895

And above result is purely tested on coco8 for simplicity, actually the same thing happened when training on the whole coco dataset.

It would be really helpful if I can get your advice on this issue, thanks in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Kevinfringe, thank you for your interest in Ultralytics 🚀! We encourage you to check out the [Docs](https://docs.ultralytics.com) to ensure you've got the best setup for your custom training. You'll find practical [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples there.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose the problem more effectively? This would typically include the script you're using and any specific configurations that can replicate the issue on our end.

For custom training ❓ Questions, it's important to include key details such as dataset examples, your configuration settings, and training logs, while ensuring you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the vibrant Ultralytics community where you can explore more opportunities for collaboration:
- For real-time discussions, hop onto [Discord](https://ultralytics.com/discord) 🎧.
- For more detailed inquiries, our [Discourse](https://community.ultralytics.com) forum might be the place for you.
- Engage with fellow developers on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

First, make sure that everything is running on the latest version of the `ultralytics` package, which might resolve existing issues. Run the following command in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Ensure you’re using a suitable and up-to-date environment. Here are some recommended options:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Follow our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon AWS**: See our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker**: For containerized environments, refer to our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/). <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge shows green, it signifies all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) checks are successful. The CI system helps verify the operation of YOLO's [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different platforms every 24 hours and upon every code commit.

This message is automatically generated, but rest assured that an Ultralytics engineer will review your issue and provide further assistance soon. 😊

**Comment by glenn-jocher** [comment]
Thank you for reaching out. It seems like you are experiencing issues with the mAP when tuning YOLO-World v2 on the COCO dataset. Here are a few steps you can consider: 

1. **Check Dataset and Annotations**: Ensure your dataset is correctly formatted and annotations are accurate. Any discrepancies can impact training results.
2. **Hyperparameter Tuning**: Experiment with different learning rates or batch sizes. Sometimes the default settings may not be optimal for your specific dataset.
3. **Pretrained Weights**: Ensure that the pretrained weights are loading correctly, as they can significantly impact initial training phases.
4. **Monitor Training Metrics**: Use tools like TensorBoard to track metrics such as loss, precision, and recall to diagnose where the model might be struggling.

Feel free to refer to the [Ultralytics documentation](https://docs.ultralytics.com/models/yolo-world/#how-do-i-train-a-yolo-world-model-on-my-dataset) for more detailed guidance. If the issue persists, providing more detailed training logs and configurations could help in further diagnosing the problem.

**Comment by Kevinfringe** [comment]
Are there any chances that you can kindly share the training log of the pretrained model? I tried the hyperparameter setting of original paper, but the mAP still rapidly decreased after a few epochs...

Thanks,

**Comment by glenn-jocher** [comment]
@Kevinfringe thank you for your question. Unfortunately, we do not share training logs for pretrained models. I recommend ensuring your dataset and settings align closely with the official documentation and experimenting with hyperparameter tuning as your dataset and environment may require adjustments. For detailed guidance, please refer to the [YOLO-World training guide](https://docs.ultralytics.com/models/yolo-world/).
